### ISSUE TITLE
[SF][Feature Request][Icon]: Icon is no longer interactive in 2.x

### ISSUE THREAD

**Issue by ksblklu02** [SAP/ui5-webcomponents#10616]
### Feature Request Description

In ui5/webcomponents 1.x, the Icon component has an "interactive" property, but in 2.x, the "interactive" property does not work anymore. This will cause a11y issue in our products because the use interactive Icon so user can navigate these icon through keyboard tab key.

### Proposed Solution

Add "interactive" property to Icon component

### Organization

SAP SuccessFactors

### Priority

Medium

### Privacy Policy

- [x] I’m not disclosing any internal or sensitive information.

**Comment by ilhan007** [comment]
In v2 there are changes in Icon's API:
https://sap.github.io/ui5-webcomponents/docs/migration-guides/to-version-2/#ui5-icon

<img width="804" alt="Image" src="https://github.com/user-attachments/assets/dc791bb5-6f96-49b2-8875-1921b8afc1f3" />
